exec/totemknet.c

```diff
@@ -1487,17 +1567,94 @@ int totemknet_reconfigure (
 	if (res) {
 		KNET_LOGSYS_PERROR(errno, LOGSYS_LEVEL_ERROR, "knet_handle_compress failed");
 	}
+	return res;
+}
+
+int totemknet_reconfigure (
+	void *knet_context,
+	struct totem_config *totem_config)
+{
+	struct totemknet_instance *instance = (struct totemknet_instance *)knet_context;
+	int res = 0;
+
+	(void)totemknet_configure_compression(knet_context, totem_config);
 
 #ifdef HAVE_LIBNOZZLE
-	/* Set up nozzle device(s). Return code is ignored, because unability
+	/* Set up nozzle device(s). Return code is ignored, because inability
 	 * configure nozzle is not fatal problem, errors are logged and
 	 * there is not much else we can do */
 	(void)setup_nozzle(instance);
 #endif
+
+	if (totem_config->crypto_changed) {
+		/* Flip crypto_index */
+		totem_config->crypto_index = 3-totem_config->crypto_index;
+		res = totemknet_set_knet_crypto(instance);
+
+		knet_log_printf(LOG_INFO, "kronosnet crypto reconfigured on index %d: %s/%s/%s", totem_config->crypto_index,
+				totem_config->crypto_model,
+				totem_config->crypto_cipher_type,
+				totem_config->crypto_hash_type);
+	}
 	return (res);
 }
 
 
+int totemknet_reconfigure_phase (
+	void *knet_context,
+	struct totem_config *totem_config,
+	uint32_t phase)
+{
+#ifdef HAVE_KNET_CRYPTO_RECONF
+	int res;
+	struct totemknet_instance *instance = (struct totemknet_instance *)knet_context;
+
+	knet_log_printf(LOGSYS_LEVEL_DEBUG, "totemknet_reconfigure_phase %d, index=%d\n", phase, totem_config->crypto_index);
+	if (phase == RELOAD_PHASE_ACTIVATE) {
```

Comment:
I would suggest using `switch` there so compiler produces warning (if enum is used) when we add/remove phase (I know, unlikely to happen, but who knows).